### PR TITLE
Update to latest PHP 8.1 syntax

### PR DESCRIPTION
**Pull request by samsonasik**
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.1, php 8.1 syntax can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
Ready for review 👍

File: test/UserRepository/HtpasswdFactoryTest/ConfigImplementingArrayAccess.php [hunk 1]
```diff
@@ -19,15 +19,11 @@
...
-
     /**
-     * @param array<string,mixed> $data
+     * @param array<array-key,mixed> $data
      */
-    public function __construct(array $data)
+    public function __construct(private array $data)
     {
-        $this->data = $data;
     }
```
Comment (gsteel) [inline test/UserRepository/HtpasswdFactoryTest/ConfigImplementingArrayAccess.php:27]: What is it here? `array-key` or `string` ?

Comment (samsonasik) [reply test/UserRepository/HtpasswdFactoryTest/ConfigImplementingArrayAccess.php:27]: on property, it defined as `array<array-key, mixed>`, on constructor param, it defined as `array<string, mixed>` so it inconsistent, I choose to use defined in property instead so psalm not show notice

**Review by gsteel — APPROVED** [review]
👍

**Pull request merged**